docs/developer-experience/artificial-intelligence.md

```diff
@@ -0,0 +1,63 @@
+
+You must sign-up for the preview [here](https://github.com/features/preview/copilot-x). Once you are accepted, you will be able to install the extension from the VS Code Marketplace. You may require the Insiders build of VS Code.
+
+## ChatGPT / Bing Chat
+
+Prior to GitHub Copilot X chat availability, we used ChatGPT and Bing Chat to provide similar functionality. We found the generation of code using ChatGPT to be most valuable for these scenarios:
+
+- __Write Code__. When using a library or framework that you are not familiar with, ChatGPT can help you get started by generating code.
+- __Build SQL Indexes__. Given a query, ChatGPT can generate a SQL index that will improve the performance of the query.
+- __Write Regular Expressions__. These are notoriously difficult to write, but ChatGPT can generate them for you if you give some sample input and describe what you want to extract.
+- __Improve and Validate__. If you are unsure of the implications of writing code a particular way, you can ask ChatGPT questions about it. For instance, you might ask if there is a way to write the code that is more performant or uses less memory. Once it gives you an opinion, you can ask it to provide documentation validating that assertion.
+- __Change Perspective__. ChatGPT can impersonate a persona or even a system and answer questions from that perspective. For example, you can ask it to explain what a particular piece of code does from the perspective of a user. You might have ChatGPT imagine it is a database adminstrator and ask it to explain how to improve a particular query.
+
+When using Bing Chat, we found much better results using Creative Mode.
+
+## Prompt Engineering
+
+Prompt Engineering is a cutting edge subject area focused on prompting strategies designed to derive the most value from an AI response.
+
+Most LLM services such as GPT-4 or Bing Chat have a limited number of prompts a user can send to it in a given amount of time. Even outside of that scenario, we want to get the most useful information as quickly as possible.
+
+This is one open source example of documentation around prompt engineering, and should be seen as a jumping off point for further investigation. [Link](https://github.com/brexhq/prompt-engineering)
+
+## Considerations
+
+It is important when using AI tools to understand how the data (including private or commercial code) might be used by the system. GitHub Copilot provides guarantees on how data will be used that should align with customer expectations, however, we should always seek a customer's consent. More information on the privacy can be found [here](https://resources.github.com/copilot-for-business/).
```

Comment:
Again here, link to the page, and let them do the guarantees. If we make statements around guarantees on this page, we bind ourselves to update this if it ever changes. Else we will have two conflicting Microsoft sources on the topic.

Reply:
This doesn't say what those guarantees are.

One problem we had with ChatGPT is we couldn't provide it any prompts that included sensitive code. However, Copilot is designed to handle that scenario. Can you provide another way to say that?